BeatTogether.DedicatedServer.Kernel/Managers/LobbyManager.cs

```diff
@@ -354,11 +209,134 @@ public void Update()
             _lastBeatmap = beatmap;
         }
 
+
+        private void CountingDown(bool isReady, float CountDownTime, bool NotStartable, bool allPlayersOwnBeatmap, BeatmapIdentifier? beatmap, GameplayModifiers modifiers)
+        {
+            // If not already counting down
+            if (CountdownEndTime == 0)
+            {
+                UpdateBeatmap(beatmap, modifiers);
+                if ((AllPlayersReady && !AllPlayersSpectating && allPlayersOwnBeatmap))
+                {
+                    CountdownEndTime = _instance.RunTime + CountdownTimeEveryoneReady;
+                    _packetDispatcher.SendToNearbyPlayers(new StartLevelPacket
+                    {
+                        Beatmap = SelectedBeatmap!,
+                        Modifiers = SelectedModifiers,
+                        StartTime = CountdownEndTime
+                    }, DeliveryMethod.ReliableOrdered);
+                }
+                else if (isReady && allPlayersOwnBeatmap)
+                {
+                    CountdownEndTime = _instance.RunTime + CountDownTime;
+                    _packetDispatcher.SendToNearbyPlayers(new SetCountdownEndTimePacket
+                    {
+                        CountdownTime = CountdownEndTime
+                    }, DeliveryMethod.ReliableOrdered);
+                }
+
+            }
+
+            // If counting down
+            else
+            {
+                // If beatmap or modifiers changed, update them
+                UpdateBeatmap(beatmap, modifiers);
+                if (CountdownEndTime <= _instance.RunTime)
+                {
+                    // If countdown just finished, send map one last time then pause lobby untill all players have map downloaded
+                    if (CountdownEndTime != -1)
+                    {
+                        _packetDispatcher.SendToNearbyPlayers(new StartLevelPacket
+                        {
+                            Beatmap = SelectedBeatmap!,
+                            Modifiers = SelectedModifiers,
+                            StartTime = CountdownEndTime
+                        }, DeliveryMethod.ReliableOrdered);
+                        CountdownEndTime = -1;
+                    }
+                    if (_playerRegistry.Players.All(p => p.GetEntitlement(SelectedBeatmap!.LevelId) is EntitlementStatus.Ok))
+                    {
+                        // sends entitlements to players
+                        _packetDispatcher.SendToNearbyPlayers(new SetPlayersMissingEntitlementsToLevelPacket
```

Comment:
Sends player entitlement to all clients once the server has received that all clients have the beatmap and starts the beatmap.
This is because when testing players were not receiving entitlements which was causing them to get stuck in the lobby infinitely in waiting for players to download screen. Now they all receive the correct entitlements and the map starts for them all. Tested and works with up to and 4 players.

Reply:
BUGFIX